vulcanai2/datasets/tabulardataset.py

```diff
@@ -319,46 +372,74 @@ def identify_unique(self, threshold):
 
     def identify_unbalanced_columns(self, threshold, non_numeric=True):
         """
-        This returns columns that are highly unbalanced, aka those that have a disproportionate amount of one value.
-        Return those columns that have at least threshold percentage of one value.
-        :param threshold: Proportion needed to define unbalanced, between 0 and 1
-        :param non_numeric: Whether non-numeric columns are also considered.
-        :return: The column list
+        This returns columns that are highly unbalanced.
+        Those that have a disproportionate amount of one value.
+
+        Parameters
+        ----------
+        threshold: Float
+            Proportion needed to define unbalanced, between 0 and 1
+        non_numeric: Boolean
+            Whether non-numeric columns are also considered.
+
+        Returns
+        -------
+        column_list: List
+            The list of column names
         """
         if non_numeric:
             columns = list(self.df.columns)
         else:
             columns = list(self.df.select_dtypes(include=np.number))
         column_list = []
         for col in columns:
-            #Check amount of null values, because if column is entirely null, max won't work.
+            # Check amount of null values, because if column is entirely null,
+            # max won't work.
             num_of_null = self.df[col].isnull().sum()
             if num_of_null != len(self):
-                col_maj = (max(self.df[col].value_counts()) / self.df[col].value_counts().sum())
+                col_maj = (max(self.df[col].value_counts()) /
+                           self.df[col].value_counts().sum())
                 if col_maj >= threshold:
                     column_list.append(col)
         return column_list
 
     def identify_highly_correlated(self, threshold):
         """
         Identify columns that are highly correlated with one-another.
-        :param threshold: Between 0 and 1. Minimum amount of correlation necessary to be identified.
-        :return: None
+
+        Parameters
+        ----------
+        threshold: Between 0 and 1.
+        Minimum amount of correlation necessary to be identified.
+
+        Returns
+        -------
+        column list: List of tuples
+            The correlation values above threshold
         """
         column_list = set()
-        featuresCorrelation = self.df.corr().abs()
-        corr_pairs = featuresCorrelation.unstack()
+        features_correlation = self.df.corr().abs()
+        corr_pairs = features_correlation.unstack()
         for index, val in corr_pairs.items():
```

Comment:
any pandas functionality that can simplify this logic? something similar to `numpy.where`?

Reply:
Since you only care about the correlations that are greater than the threshold. you can use the .where to simply extract those instead of setting the others to 0 and then loop through again. does that make sense? you could probably reduct this to only one line

Reply:
@rfratila I don't know if that's possible for dataframe, because it would require removing entire rows and columns. For example, in the attached image in order to remove the zero from that row+column combo we'd have to either remove the entire row or the entire column. We don't want to do that because you can see that some highly correlated values would be lost if we removed. Does that make sense?
![corr_example_vulc2](https://user-images.githubusercontent.com/31524161/49392858-86d6ca80-f6e4-11e8-8200-c5a47a16fa7d.PNG)

Reply:
could you create a temp data frame that collects the correct rows instead of having to remove them from the original if that's not feasible? I'd just like to loop at most once to do this

Reply:
@rfratila in the current implementation we are only looping once

Reply:
ok great!